host/tests/dsDisplay_L3_Tests/dsDisplay_test01_GetDisplayEDID.py

```diff
@@ -0,0 +1,181 @@
+
+        # Test Setup configuration file
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Session for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+
+        # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.deviceProfile = dir_path + "/" + self.cpe.get("test").get("profile")
+
+    def testDownloadAssets(self):
+        """
+        Downloads the artifacts in test-setup configuration file to the dut.
+
+        Args:
+            None.
+        """
+
+        self.deviceDownloadPath = self.cpe.get("target_directory")
+
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+
+        # Download test artifacts to device
+        url = test.get("artifacts")
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+
+    def testRunPrerequisites(self):
+        """
+        Runs Prerequisite commands listed in test-setup configuration file on the dut.
+
+        Args:
+            None.
+        """
+
+        #Run test specific commands
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+        cmds = test.get("execute")
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+    #TODO: Current version supports only manual verification.
+    def testGetEDID(self, manual=False, port="dsVIDEOPORT_TYPE_HDMI" , index:int = 0):
+        """
+        Gets the EDID of the display.
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                    Defaults to other verification methods
+                                    port (str, optional): The type of video port to retrieve the EDID.
+                                    index (int, optional): The index of the port to retrieve the EDID.
+        Returns:
+            bool
+        """
+        # Get the EDID from the display
+        edid = self.testdsDisplay.getEdid(port, index)
+
+        if manual:
+            # Manual verification
+            return self.testUserResponse.getUserYN(f"Is the EDID {edid} correct? (Y/N):")
+        else:
+            # Automation verification
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        This function will test the Display by getting the EDID of the display.
+
+        This function:
+        - Downloads the required assets.
+        - Runs the prerequisite commands.
+        - Retrieves EDID for each supported port and verifies them.
+        - Cleans up assets after the test.
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        # Download the assets listed in the test setup configuration file
+        self.testDownloadAssets()
```

Comment:
Move download functions to testPrepareFunction which is called once per test. refer https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/feature/gh233_dsAudio_L3_python_updates/host/tests/L3_TestCases/dsAudio/dsAudioHelperClass.py#L132